src/psyclone/psyir/backend/fortran.py

```diff
@@ -601,34 +601,53 @@ def gen_decls(self, symbol_table, args_allowed=True):
             which does not have a GlobalInterface or LocalInterface as this \
             is not supported by this backend.
         :raises VisitorError: if args_allowed is False and one or more \
-                              argument declarations exist in symbol_table.
-        :raises VisitorError: if any symbols representing variables (i.e. \
-            not kind parameters) without an explicit declaration or 'use' \
-            are encountered.
+            argument declarations exist in symbol_table.
+        :raises VisitorError: if there are any symbols in the supplied table \
+            that do not have an explicit declaration and there are no \
+            wildcard imports.
 
         '''
         declarations = ""
-
-        if not all([isinstance(symbol.interface, (LocalInterface,
-                                                  GlobalInterface))
-                    for symbol in symbol_table.symbols
-                    if isinstance(symbol, RoutineSymbol)]):
-            raise VisitorError(
-                "Routine symbols without a global or local interface are not "
-                "supported by the Fortran back-end.")
+        # Keep a record of whether we've already checked for any wildcard
+        # imports to save doing so repeatedly
+        wildcard_imports_checked = False
+        has_wildcard_import = False
+
+        routine_symbols = [symbol for symbol in symbol_table.symbols
+                           if isinstance(symbol, RoutineSymbol)]
+        for sym in routine_symbols:
+            if (not isinstance(sym.interface, (LocalInterface,
```

Comment:
The test that checks this uses:
```
# Now add a user-defined routine symbol
symbol_table.add(RoutineSymbol("arg_sub", interface=ArgumentInterface()))
```
This surprised me as Fortran would need to declare an interface to pass a routine as argument, and thebackend doesn't do anything like this and will probably produce incorrect code. Also this isinstance could just check for UnresolvedInterface I think. 

So would it make sense to update to the following code?
```
for sym in routine_symbols:
    if (isinstance(sym.interface, UnresolvedInterface) and sym.name.upper() not in FORTRAN_INTRINSICS):
        # Current loop body
    else if isinstance(sym.interface, ArgumentInterface):
        # Fails, not supported
```

Reply:
Oh yes, that's a good point. I've done as you suggest and extended the test appropriately.